internal/filter/filter.go

```diff
@@ -169,6 +211,67 @@ func (f *Filter) FindModulesForNode(ctx context.Context, node client.Object) []r
 	return reqs
 }
 
+// FindModulesForNMCNodeChange finds the modules that are affected by node changes that result
+// in ModuleNMCReconcilerNodePredicate predicate. First it find all the Module that can run on the node, based
+// on the Modules' Selector field and on node's labels. Then, in case NMC for the node exists, it adds all the
+// Modules already set in NMC ( in case they were not added in a previous step).
+func (f *Filter) FindModulesForNMCNodeChange(ctx context.Context, node client.Object) []reconcile.Request {
+	logger := ctrl.LoggerFrom(ctx).WithValues("node", node.GetName())
+
+	reqs := make([]reconcile.Request, 0)
```

Comment:
I think we can remove that variable since it's now unused.

Reply:
fixed